n3fit/src/evolven3fit_new/evolve.py

```diff
@@ -0,0 +1,221 @@
+import pathlib
+import logging
+import sys
+
+import numpy as np
+from reportengine.compat import yaml
+
+from ekobox import genpdf, gen_info
+from ekomark import apply
+from eko import basis_rotation as br
+from eko import output
+from validphys.api import API
+
+from . import utils, eko_utils
+
+
+_logger = logging.getLogger(__name__)
+
+LOG_FILE = "evolven3fit_new.log"
+
+LOGGING_SETTINGS = {"formatter" : "%(asctime)s %(name)s/%(levelname)s: %(message)s",
+                    "level" : logging.INFO
+}
+
+def evolve_fit(
```

Comment:
About this, there has already been a discussion with no proper conclusion so I decided to wait. Anyway, even merging this as it is, we can always change it afterwards.

Reply:
True, but in reality that won't happen so maybe it's better to resolve that discussion now

Reply:
This is also true. So, in general I don't see anything wrong in using the API here in that this is actually a sort of external application. Moreover, at least for me, it is easier to mantain as it is. At the end I am using the API only to get the theoryID (just a line of code) so using vp functionalities won't be much of an improvement I think (but it could make everything less clear for a new mantainer for example)

Reply:
As discussed, one should prefer the least magic option when doing things. Especially when we are arguing about doing

```
In [6]: from validphys.loader import Loader

In [7]: Loader().check_theoryID(200)
Out[7]: TheoryIDSpec(id=200, path=PosixPath('/home/zah/anaconda3/envs/nnpdfdev/share/NNPDF/data/theory_200'))
```

The API makes it less clear what is going on (you cannot control+click in your IDE ans see what function is being called) and encodes some defaults which could (should) be configurable here (specifically having --net/--no-net options).

Reply:
> As discussed, one should prefer the least magic option when doing things

About this I perfectly agree: it would be wonderful to have a library that does not require a magic `API` at all...